cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/writer/MessagingLineageWriter.java

```diff
@@ -81,14 +84,27 @@ private void publishLineage(ProgramRunId programRunId, DataAccessLineage lineage
   }
 
   private void publish(MetadataMessage message) {
-    StoreRequest request = StoreRequestBuilder.of(topic).addPayload(GSON.toJson(message)).build();
-    try {
-      Retries.callWithRetries(() -> messagingService.publish(request), retryStrategy, Retries.ALWAYS_TRUE);
-    } catch (Exception e) {
-      LOG.trace("Failed to publish metadata message: {}", message);
-      ProgramRunId programRunId = (ProgramRunId) message.getEntityId();
-      throw new RuntimeException(String.format("Failed to publish metadata message of type '%s' for program " +
-              "run '%s'.", message.getType(), programRunId), e);
+    String messageJson = GSON.toJson(message);
+    int lineageSize = messageJson.length();
+
+    if (lineageSize > publishSizeLimit) {
+      if (!isLineageWarningLogged) {
+        LOG.warn("Some lineage messages are not being logged as they are larger than the limit of {} bytes.",
```

Comment:
For future reference, you can use `Loggers.sampling(LOG, LogSamplers.onceEvery(Integer.MAX_VALUE))` to help.